core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.spec.ts

```diff
@@ -422,19 +414,140 @@ describe('Contributor dashboard Admin page', () => {
       expect(component.selectedLanguage.id).toBe(nonDefaultLanguage.id);
     }));
 
-    it('should select last activity from dropdown', fakeAsync(() => {
+    it(
+      'should initially filters users by whether their last activity occurred ' +
+        'within the last 90 days',
+      fakeAsync(() => {
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+
+        expect(component.filter.numberOfDaysSinceLastActivityStartDate).toEqual(
+          0
+        );
+
+        expect(component.filter.numberOfDaysSinceLastActivityEndDate).toEqual(
+          90
+        );
+      })
+    );
+
+    it("should changes filter by users' last activity when start date changes", fakeAsync(() => {
+      const numberOfDaysSinceFirstActivity = 7;
       component.ngOnInit();
       tick();
       fixture.detectChanges();
 
-      let lastActivityDays = 7;
-      component.selectLastActivity(lastActivityDays);
-      expect(component.selectedLastActivity).toEqual(lastActivityDays);
-      lastActivityDays = 0;
-      component.selectLastActivity(lastActivityDays);
-      expect(component.selectedLastActivity).toEqual(lastActivityDays);
+      component.changeFirstActivity(
+        component.getDateThatIsDaysBeforeToday(numberOfDaysSinceFirstActivity)
+      );
+
+      expect(component.filter.numberOfDaysSinceLastActivityStartDate).toEqual(
+        numberOfDaysSinceFirstActivity
+      );
     }));
 
+    it(
+      "should not change filter by users' last activity when given start date " +
+        'is in the future',
+      fakeAsync(() => {
+        const numberOfDaysSinceFirstActivity = -1;
+        component.ngOnInit();
+        tick();
+        fixture.detectChanges();
+        const initialFirstActivity =
+          component.filter.numberOfDaysSinceLastActivityStartDate;
+
+        component.changeFirstActivity(
+          component.getDateThatIsDaysBeforeToday(numberOfDaysSinceFirstActivity)
+        );
+
+        expect(component.filter.numberOfDaysSinceLastActivityStartDate).toEqual(
+          initialFirstActivity
+        );
+      })
+    );
+
+    it(
+      "should not change filter by users' last activity when given start date " +
+        "comes before filter's current end date",
```

Comment:
`after`, not `before`. The filter's start date should always come before the filter's end date.

Reply:
right, done